Converts remaining vectorial particle props to Vector3d

RudolfWeeber opened the pull request.

RudolfWeeber committed:
Convert more particle properties to Vector3d

RudolfWeeber committed:
Last bunch of vectorial particle properties to Vector3d

RudolfWeeber committed:
Core: Fix H5md with  Vector<> properties

RudolfWeeber committed:
Core: Temporary solution for rinertia property when RINERITA not compiled in

fweik commented on src/core/particle_data.hpp:
```diff
@@ -104,19 +104,19 @@ struct ParticleProperties {
 #ifdef AFFINITY
   /** parameters for affinity mechanisms */
-  double bond_site[3] = {-1., -1., -1.};
+  Vector3d bond_site[3] = {-1., -1., -1.};
```
This should not be an array of Vectors.

RudolfWeeber committed:
Fix affinity. Re-add missing AffinityInteraciton in py

from https://raw.githubusercontent.com/icimrak/espresso/92f783f164daf88142943ec7fa205b1d0e708f1e/src/python/espressomd/interactions.pyx

I don't know where it went missing. This was undetected, since AFFINITY was not part of Maxset.

RudolfWeeber committed:
Add AFFINITY to maxset

RudolfWeeber commented:
AFFINITY was not part of maxset, so a wider regression was not noticed.

I had to re-add the Affinity Interactoin Python class in interactions.pyx from an older version.
I do not know, when it went missing.
@icimrak, would you be willing to contribute a testcase for AFFINITY and for MEMBRANE_COLLISOIN?
It compiles now, but I have no way of telling, whether it still does what it should.

RudolfWeeber committed:
Core: Remove some consts to make mpi happy

RudolfWeeber committed:
More const issues, np.array -> array-locked for bond_site
